src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -414,4 +423,37 @@ private static synchronized void deleteEventToJenkinsJobRun(
         // clean up
         innerDeleteEventToJenkinsJobRun(build);
     }
+
+  /**
+   * Reconciles Jenkins job runs and OpenShift builds
+   *
+   * Deletes all job runs that do not have an associated build in OpenShift
+   */
+  private static synchronized void reconcileRunsAndBuilds() {
+    logger.info("Reconciling job runs and builds");
+
+    List<WorkflowJob> jobs = Jenkins.getActiveInstance().getAllItems(WorkflowJob.class);
+
+    for (WorkflowJob job : jobs) {
+      BuildConfigProjectProperty bcpp = job.getProperty(BuildConfigProjectProperty.class);
+      BuildList buildList = getAuthenticatedOpenShiftClient().builds()
+        .inNamespace(bcpp.getNamespace()).withLabel("buildconfig=" + bcpp.getName()).list();
```

Comment:
you've got to check for null before accessing `bcpp` ... there you are examining all jobs, it is not a given that the `BuildConfigProjectProperty` will be there

Reply:
updated in next push